src/cljs/main/broadfcui/page/profile.cljs

```diff
@@ -38,21 +38,10 @@
 (defn get-url-search-param [param-name]
   (.get (js/URLSearchParams. js/window.location.search) param-name))
 
-(defn decode-base64-json [b64string]
-  (utils/parse-json-string (js/window.atob b64string) true))
-
-(defn encode-base64-json [map]
-  (js/window.btoa (utils/->json-string map)))
-
-(defn get-fence-oauth-href [oauth_host client_id provider]
-  (str oauth_host
-       "/user/oauth2/authorize?response_type=code&client_id="
-       client_id
-       "&scope=openid+google_credentials&redirect_uri="
-       (js/encodeURIComponent
-        (let [loc js/window.location]
-          (str (.-protocol loc) "//" (.-host loc) "/#fence-callback")))
-       "&state=" (encode-base64-json {:provider provider})))
+(defn auth-url-link [href text]
+  (if href
+    (links/create-external {:href href :target "_self"} text)
+    (spinner {:ref "pending-spinner"} "Getting link information...")))
```

Comment:
[String refs are deprecated](https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs), but given that there are lots of others you could conceivably leave this in. Just something to be aware of.